dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestUUIDEntityIDServlet.java

```diff
@@ -480,4 +525,49 @@ public void testBasicUUIDIdentity_UUID_Generator_JTA_XML() {
             }
         }
     }
+
+    /**
+     * Verify that an entity using a UUID type for its identity can use the UUID generator for primary key generation. XML Variant. Merge
+     */
+    @Test
+    public void testBasicUUIDIdentity_UUID_Generator_Merge_JTA_XML() {
+        EntityManager tem = null;
+
+        try {
+            tem = emfJta.createEntityManager();
+            Assert.assertNotNull(tem);
+            Assert.assertTrue(tem.isOpen());
+
+            XMLUUIDUUIDGenEntity entity = new XMLUUIDUUIDGenEntity();
+            entity.setStrData("Some string data");
+
+            tx.begin();
+            tem.joinTransaction();
+            tem.merge(entity);
```

Comment:
When using "merge", it would be more correct to use the return value, so please update this to:
entity = tem.merge(entity);